src/main/java/net/minecraftforge/client/ItemOverlayHandler.java

```diff
@@ -15,76 +18,72 @@
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.EnumFacing;
-import net.minecraftforge.common.ForgeModContainer;
+import net.minecraftforge.fml.common.registry.RegistryDelegate;
 
 /**
  * Offshore handler for the Items' extra passes. Specifically for reducing patch size & maintaining readability.
  * */
 public final class ItemOverlayHandler
 {
 
-    private static final java.util.Map<net.minecraftforge.fml.common.registry.RegistryDelegate<Item>, IItemOverlay> itemOverlayMap = com.google.common.collect.Maps.newHashMap();
+    /**
+     * Map of active Item.delegate > IItemOverlay entries.
+     * */
+    private static final Map<RegistryDelegate<Item>, IItemOverlay> itemOverlayMap = Maps.newHashMap();
 
+    /**
+     * Number of passes to be run.
+     * */
     private static final byte NUM_PASS = 2;
 
     private ItemOverlayHandler(){}
 
     /**
-     * Handles the execution of overlays.
-     * 
-     * @param textureManager the current TextureManager
-     * @param stack the stack in question
-     * @param model 
-     * 
-     * @return true to render the item again, and false to end the pass loop.
+     * Handles the execution of overlays. This feeds from the {@value itenOverlayMap} for custom IITemOverlays to be run in lieu of the encapsulated vanilla
+     *  methodology. 
+     *  
+     * @return returns false canceling vanilla's default overlay for minimal patch size
      * */
-    public static boolean run(TextureManager textureManager, ItemStack stack, IBakedModel model)
+    public static boolean applyForgeOverlay(TextureManager textureManager, ItemStack stack, IBakedModel model)
```

Comment:
@mezz  adding a flag here to optionally prevent state changes would be necessary for batching in JEI right?